packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -40,7 +40,7 @@ class RenderProxyBox extends RenderBox with RenderObjectWithChildMixin<RenderBox
   /// Proxy render boxes are rarely created directly because they simply proxy
   /// the render box protocol to [child]. Instead, consider using one of the
   /// subclasses.
-  RenderProxyBox([RenderBox child = null]) {
+  RenderProxyBox([RenderBox child = null]) { //ignore: avoid_init_to_null
```

Comment:
I don't think this is inference related.  I think it is related to strong mode override checking.  The fact that adding and removing `= null` changes the error looks like a bug to me.